src/app/pages/storage/volumes/datasets/dataset-form/dataset-form.component.ts

```diff
@@ -616,20 +602,40 @@ export class DatasetFormComponent implements Formconfiguration{
         entityForm.formGroup.controls['recordsize'].setValue('INHERIT');
         }
         else {
-          this.ws.call('pool.dataset.query', [[["id", "=", this.parent]]]).subscribe((parent_dataset)=>{
+          this.ws.call('pool.dataset.query', [[["id", "=", this.parent]]]).subscribe((parent_dataset)=>{      
             this.parent_dataset = parent_dataset[0];
-            if (parent_dataset[0].quota.rawvalue === '0') {
-              entityForm.formGroup.controls['quota_unit'].setValue('B');
-            }
-            if (parent_dataset[0].refquota.rawvalue === '0') {
-              entityForm.formGroup.controls['refquota_unit'].setValue('B');
-            }
-            if (parent_dataset[0].reservation.rawvalue === '0') {
-              entityForm.formGroup.controls['reservation_unit'].setValue('B');
-            }
-            if (parent_dataset[0].refreservation.rawvalue === '0') {
-              entityForm.formGroup.controls['refreservation_unit'].setValue('B');
-            }
+            const current_dataset = _.find(this.parent_dataset.children, {'name':this.pk});
+            const lower_recordsize_map = {
+              '512':'512',
+              '1K':'1K',
+              '2K':'2K',
+            }; 
+            if ( current_dataset.hasOwnProperty("recordsize")) {
```

Comment:
Why not just an && instead of an additional nested if on all of these?  Short-circuiting should prevent trying to set the value of a property that does not exist.

Reply:
addressed.